### PR TITLE
Fixed text/font issues when mixing unicode and bytes in python 3

### PR DESCRIPTION
**Pull request by jdeschenes**
* Added preliminary test module for text

I am unsure how best to test for text based data. Let me know what you think

### CODE REVIEW

File: celiagg/tests/test_text.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import celiagg as agg
+
+
+def test_font_byte_string():
+    canvas = agg.CanvasG8(np.zeros((1, 1), dtype=np.uint8))
+    gs = agg.GraphicsState()
+    transform = agg.Transform()
+
+    text_unicode = "Hello!"
```
Comment (jwiggins) [inline celiagg/tests/test_text.py]: Any special reason why this isn't `text_unicode = u"Hello!"`? Seems like without that, Unicode text isn't getting tested on Python 2.7.

File: celiagg/tests/test_text.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    font_unicode = agg.Font("Times New Roman", 12.0, agg.FontCacheType.RasterFontCache)
```
Comment (jwiggins) [inline celiagg/tests/test_text.py]: Same question here

**Comment by jwiggins** [comment]
I'm not clear what you mean by "test for text based data". Do you mean the type check in `_get_utf8_text()`? I vaguely remember being unclear about the check there when I wrote it.

**Comment by jdeschenes** [comment]
I simply mean tests to check that writing text on the canvas is operating properly. From the unit test, we don't know:
* Is the text "Hello"written properly?
* What happens if the font change? Does it make any difference at all?
* Does it support unicode character at all or simply ascii? Japanese for example? Emoticons, etc.

File: celiagg/tests/test_text.py (hunk 1 above)
Comment (jdeschenes) [inline celiagg/tests/test_text.py]: Whoops, fixed in latest push.

**Comment by jwiggins** [comment]
Ahhh... Yeah. That's a hard problem for text rasterization. I've only manually tested that it actually renders text correctly.

I know for a fact that it renders Chinese correctly, so I would assume the other CJK languages work as long as an appropriate font is used. It would be nice to test Korean though, because Emoji does not work. I believe both map multiple code points to single glyphs.

**Comment by jwiggins** [comment]
Thanks for the fix! Merging.

**Pull request merged**